docs/bridging/bridging-fa-how.md

```diff
@@ -0,0 +1,97 @@
+---
+
+The process of bridging FA tokens is similar to the process of bridging tez.
+In short, the bridge uses tickets to send tokens from the source network to the target network.
+
+## Contracts
+
+The bridging process relies on smart contracts that convert tokens to [tickets](https://docs.tezos.com/smart-contracts/data-types/complex-data-types#tickets) and transfer the tickets between Tezos and Etherlink.
+These contracts are an implementation of the [TZIP-029](https://gitlab.com/baking-bad/tzip/-/blob/wip/029-etherlink-token-bridge/drafts/current/draft-etherlink-token-bridge/etherlink-token-bridge.md) standard for bridging between Tezos and Etherlink.
+
+Each FA token needs its own copy of these contracts to be able to bridge the token:
+
+- **Ticketer contract**: Stores tokens and issues tickets that represent them
+- **Token bridge helper contract**: Accepts requests to bridge tokens on layer 1, uses the ticketer contract to get tickets for them, and sends the tickets to Etherlink
+- **ERC-20 proxy contract**: Stores tickets and mints ERC-20 tokens that are equivalent to the FA tokens in layer 1
+
+Examples of these contracts and tools to deploy them are available in the repository https://github.com/baking-bad/etherlink-bridge.
+
+## Depositing tokens from layer 1 to Etherlink
+
+The process of bridging FA-compatible tokens from layer 1 to Etherlink (also known as depositing tokens) follows these general steps:
+
+1. A Tezos user gives the token bridge helper contract access to their tokens.
+
+   - For FA1.2 tokens, the user gives the helper contract an allowance of tokens.
+   - For FA2 tokens, the user makes the helper contract an operator of their tokens.
+
+   For information about token access control, see [Token standards](https://docs.tezos.com/architecture/tokens#token-standards) on docs.tezos.com.
+
+1. The user calls the helper contract's `deposit` entrypoint.
+The request includes the amount of tokens to bridge, the address of the Etherlink Smart Rollup, and the user's Etherlink wallet address, but not the tokens themselves.
+
+1. The token bridge helper contract stores the address of the Etherlink Smart Rollup and the user's Etherlink address temporarily.
+
+1. The helper contract (as an operator of the user's tokens or with an allowance of the user's tokens) calls the token contract to transfer the tokens from the user's account to its account.
+
+1. The helper contract calls the ticketer contract's `deposit` entrypoint and includes the tokens.
+
+1. The ticketer contract stores the tokens and creates a [ticket](https://docs.tezos.com/smart-contracts/data-types/complex-data-types#tickets) that represents the receipt of the tokens.
+
+1. The ticketer contract sends the ticket back to the helper contract.
+
+1. The helper contract forwards the ticket to the Smart Rollup inbox and clears its storage for the next transfer.
+
+1. The Etherlink Smart Rollup receives the ticket and sends it to the ERC-20 proxy contract.
```

Comment:
Hmmm. Of course the Tezos contracts are supposed to store the ERC20 address. Here's an example token bridge helper contract, with the ERC20 address in bytecode in the storage:
https://ghostnet.tzkt.io/KT1PaqqmLgyUKyrWQG9tP57rt6tag8nGSrMh/storage/

So maybe you're describing an alternate way of handling the deposit, where instead of sending the ticket to the ERC20 proxy, we send the ticket directly to the user's Etherlink account? Similarly, on item 2 the ERC20 proxy call fails and the ticket goes to the user's Etherlink account.

I can't really say if those scenarios are worth documenting -- who can we ask? Is there info somewhere on how tickets behave in Etherlink user accounts?

Reply:
> I can't really say if those scenarios are worth documenting -- who can we ask?

At the very least it's not strictly required for the first version.

> Is there info somewhere on how tickets behave in Etherlink user accounts?

I'm sure there is, I cannot find it atm but I can give insights as I know the implementation on Etherlink side.

Tickets do not really belong to Etherlink user accounts. Etherlink has a ticket table, where the **actual** owner is itself, Etherlink. The ticket table is basically a ledger where you can find how many tickets (based on its hash) belongs to an Ethereum address (EOA or contract). 

When you deposit tickets, there can be 3 cases:
1. Tickets are transferred to Proxy ERC20, tickets belongs to the contract.
1. Call to Proxy ERC20 fails (during the deposit), tickets belongs to the EOA. 
1. Proxy ERC20 is not provided, tickets belongs to the EOA.

That means that if you want to withdraw these tickets there are 2 possibilities:
1. The tickets belongs to the Proxy ERC20 contract, you have to "request" the withdrawal to the contract (which is done automatically in the precompiled contract).
1. The tickets belongs to the EOA, you don't need any proxy contract.

None of this is really important. The main information is that if by mistake the deposit to ERC20 contract fails, the assets are not lost, the ownership fallbacks to the supposed receiver.

Reply:
Ok, I have created a follow-up task to cover ticket behavior and these details after we get the initial FA bridge docs merged.